### DISCUSSION TITLE
Contributing Themes / CSS

### DISCUSSION THREAD

**Discussion by nvlang** [inventree/InvenTree#8474] [Q&A] [answered]
Hi there! Thank you for an amazing project! I have a few questions:

- Are contributions to the styling of the frontend welcome? Esp. given that a lot of that kind of stuff is rather subjective.
- I think I saw somewhere that, for the old UI, themes could be contributed without changing defaults. For the new UI, that doesn't seem to be possible, given that theme selection is just toggling between dark and light mode. Am I mistaken?
- I'm a bit confused about where the styles for the new UI are defined — is it just [`src/frontend/src/main.css.ts`](https://github.com/inventree/InvenTree/blob/113b9e9fd9b315ea96a1ee869185203976f92f29/src/frontend/src/main.css.ts) and styles defined directly in `.tsx` files?
- How are styles for the mobile apps kept in sync, if at all?

I'd love to tweak the styling for my own instance and contribute my work to the project, but I also don't want to step on anyone's toes.

Thank you for your time!

**Comment by SchrodingersGat** [comment, marked as answer]
Hi @nvlang thanks for reaching out!

> Are contributions to the styling of the frontend welcome? Esp. given that a lot of that kind of stuff is rather subjective.

Yes, absolutely. We have been building up the functionality of the new user interface but it could definitely use some attention from someone with more UI/UX experience. 

> For the new UI, that doesn't seem to be possible, given that theme selection is just toggling between dark and light mode. Am I mistaken?

There is some configurability around base colors, in the "user settings" panel. But that is about it. The granular configurability via custom css (like the old UI) is not supported.

> I'm a bit confused about where the styles for the new UI are defined — is it just [src/frontend/src/main.css.ts](https://github.com/inventree/InvenTree/blob/113b9e9fd9b315ea96a1ee869185203976f92f29/src/frontend/src/main.css.ts) and styles defined directly in .tsx files?

We lean heavily on the styling provided by [mantine.dev](https://mantine.dev/). This means that currently the UI looks a lot like a generic mantine site. Not terrible, but it would be good to have a more consistent and distinctive "look and feel".

> How are styles for the mobile apps kept in sync, if at all?

The mobile app is separate, that is written in flutter, and there is no effort (to this stage) to keep the styling in step between the user interfaces. That would be nice! But, these things take developer time which we are short on.

If you have some ideas and time to contribute, I'd be happy to chat further!

**Comment by SchrodingersGat** [comment]
@nvlang have you had a chance to look at this yet?

**Reply by nvlang** [reply]
Hi, thanks for checking in! I haven't yet, and unfortunately, it's proven harder to find the time to contribute than I had originally hoped, so I'm afraid I can't make any commitments at the moment. I hope that I might be able to contribute further down the line, but until then, I'm sorry to disappoint :(

**Reply by SchrodingersGat** [reply]
@nvlang no drama - let me know when you have some time!
